docs/RELEASE-CHECKLIST.md

```diff
@@ -0,0 +1,68 @@
+# Release Checklist
+
+This checklist is designed to ensure a smooth and successful release process for the project. Each item should be completed before the release is finalized.
+
+- Release:
+  - [ ] Create a release branch of the form 'release-vMAJOR.MINOR'
+  - [ ] Modify the version variables on that branch to:
+    - Remove the pre-release portion (if any)
+    - Set the build metadata to 'release.local'
+    - Example: 'Version = "1.0.4+release.local"'
+      - See [Version Update](#version-update) section below for details
+        on updating version variables.
+  - [ ] Update the version varible(s) on the master branch to the next
+        expected version while retaining a pre-release of 'pre'
+    - [ ] Tag the release in Git of the form 'vMAJOR.MINOR.PATCH'
+    - [ ] Push the changes to the remote repository
```

Comment:
Are these to intended to be indented more than others?

Reply:
Yes?

- Task Title
   - [ ] Task item list